cmake/packaging.cmake

```diff
@@ -278,16 +295,12 @@ function(generatePackageTarget)
     set(CPACK_STRIP_FILES ON)
   endif()
 
-  if(CPACK_GENERATOR STREQUAL "TGZ")
-    set(CPACK_INCLUDE_TOPLEVEL_DIRECTORY 0)
-    set(CPACK_SET_DESTDIR ON)
-  endif()
-
   if(DEFINED PLATFORM_LINUX)
     set(OSQUERY_PACKAGE_RELEASE "1.linux")
-
     if(CPACK_GENERATOR STREQUAL "TGZ")
       set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}_${OSQUERY_PACKAGE_RELEASE}_x86_64")
+      set(CPACK_INCLUDE_TOPLEVEL_DIRECTORY 0)
+      set(CPACK_SET_DESTDIR ON)
```

Comment:
TGZ should work on macOS too (the packaging is run on the CI, the contents not yet tested though).
Also,  probably we should use ZIP so that's usable by all platforms.

I think there are three choices here:
1) Keep it as it was before
2) Keep the change but add proper TGZ support on macOS
3) Remove TGZ support but add ZIP, for all the platforms